src/Effects/EffectMgr.cpp

```diff
@@ -117,6 +120,98 @@ static const rtosc::Ports local_ports = {
                 }
             }
         }},
+    {"numerator::i", rShort("num") rDefault(0) rLinear(0,99)
+        rProp(parameter) rDoc("Numerator of ratio to bpm"), NULL,
+        [](const char *msg, rtosc::RtData &d)
+        {
+            EffectMgr *eff = (EffectMgr*)d.obj;
+            if(rtosc_narguments(msg)) {
+                int val = rtosc_argument(msg, 0).i;
+                if (val) {
```

Comment:
The range is 0..99 right now, so val being zero would appear to be valid. As such it should be if(val >= 0). If the zero case isn't intentional feel free to modify the condition as needed.

Reply:
factor 0 makes no sense. maybe i change the range.

Reply:
it makes only sense as default because the freq/delay is set via preset.